include/xsimd/arch/xsimd_avx.hpp

```diff
@@ -881,6 +881,14 @@ namespace xsimd
             return _mm256_round_pd(self, _MM_FROUND_TO_NEAREST_INT);
         }
 
+        // nearbyint_as_int
+        template <class A>
+        inline batch<int32_t, A> nearbyint_as_int(batch<float, A> const& self,
+                                                  requires_arch<avx>) noexcept
+        {
+            return _mm256_cvtps_epi32(self);
```

Comment:
ok, so it took me time to gather my thoughts on that topic : I was wondering how this code was different from a regular *fast_cast* and it's because it's not doing the same kind of rounding as ``_mm256_cvttps_epi32``. Let's got for ``nearbyint_as_int``